### PR TITLE
add workflow for registering TRE bundle into a TRE environment

### PR DESCRIPTION
**Pull request by guybartal**
# Add workflow for building, publishing and registering TRE bundle into a TRE environment

## What is being addressed

This PR adds a workflow that will build, publish and register a TRE bundle into a TRE environment.

## How is this addressed

- it runs the correct make register command inside devcontainer action according to the provided workflow parameters.
- by default, if not provided, it will build, publish and register the base workspace

### CODE REVIEW

File: .github/workflows/register_tre_bundle.yml [hunk 1]
```diff
@@ -0,0 +1,270 @@
+---
+name: Register TRE Bundle
+
+# This workflow is intended to be used to register a TRE bundle into an existing TRE environment
+# by default, if not specified, the base workspace bundle will be registered into the TRE environment.
```
Comment (yuvalyaron) [inline .github/workflows/register_tre_bundle.yml:5]: wouldn't it be clearer to throw an error saying the argument is missing?

File: .github/workflows/register_tre_bundle.yml [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+on:
+  workflow_call:
+    inputs:
+      bundle_type:
+        description: The type of bundle to register
+        type: string
+        required: true
+  push:
+    branches:
+      - guybartal/publish-bundle-workflow
```
Comment (yuvalyaron) [inline .github/workflows/register_tre_bundle.yml:16]: mistake?

**Review by yuvalyaron — CHANGES_REQUESTED** [review]

File: .github/workflows/register_tre_bundle.yml (hunk 1 above)
Comment (guybartal) [reply .github/workflows/register_tre_bundle.yml:5]: I agree, done

File: .github/workflows/register_tre_bundle.yml (hunk 2 above)
Comment (guybartal) [reply .github/workflows/register_tre_bundle.yml]: removed along with other triggers

**Review by yuvalyaron — APPROVED** [review]

**Pull request merged**